x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/LogicalPlanBuilder.java

```diff
@@ -330,11 +334,31 @@ public Sequence visitSequence(SequenceContext ctx) {
             until = defaultUntil(source);
         }
 
+        if (maxSpan.duration() < 0 && queries.stream().anyMatch(x -> x.missingEvent())) {
+            throw new ParsingException(
+                source,
+                "No maxspan defined. Maxspan is mandatory for sequences with missing events",
+                queries.size()
+            );
```

Comment:
The style of parsing error message is a bit different. I've attempted a rewording for this one. Also `queries.size()` is not needed.

```suggestion
            throw new ParsingException(
                source,
                "[maxspan] is required for sequences with missing events queries; found none"
            );
```